spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java

```diff
@@ -100,31 +110,69 @@
 
 	private AtomicInteger failed;
 
-	private ImapServer imapIdleServer;
+	private GreenMail imapIdleServer;
+	private GreenMailUser user;
 
 	@Autowired
 	private ApplicationContext context;
 
+	private static class ImapSearchLoggingHandler extends Handler {
+		private final List<String> searches = new ArrayList<>();
+		private final List<String> stores = new ArrayList<>();
+		private static final String SEARCH = " SEARCH ";
+		private static final String STORE = " STORE ";
+
+		@Override
+		public void publish(LogRecord record) {
+			if ("com.sun.mail.imap.protocol".equals(record.getLoggerName())) {
```

Comment:
I see you are making some "hard core" logic to catch and analyze mail logs. If there is no other way to implement our tests, can we consider to do that without extra SLF in the middle?
Looks like we just can bridge JUL to Log4J directly: https://stackoverflow.com/questions/21781026/how-to-send-java-util-logging-to-log4j2

Reply:
Yeah, it's a bit weird. The tests used state of the old TestMailServer to assert parts of a query. I wanted to replicate that. It's not the nicest code. :)
I'll look into the log4j-redirect. Never used it, and didn't word=k on first try, because JUnit initialises JUL "too early". Is there anywhere in the codebase a test that asserts in some log output I could steal?

Reply:
OK. No problem. Perhaps you are right and we need to revise the logic of this test instead of trying to find some struts.

Well, don't assert log outputs, but rather "spy" on the logs if we indeed need to assert against some state which is available only via logging.

See something like this pattern:
```
LogAccessor logger = spy(TestUtils.getPropertyValue(channel, "logger", LogAccessor.class));
when(logger.isDebugEnabled()).thenReturn(true);
new DirectFieldAccessor(channel).setPropertyValue("logger", logger);
...
ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
verify(logger, times(2)).debug(captor.capture());
List<String> logs = captor.getAllValues();
```
The class I took this snippet is `DirectChannelTests`